Tests/ML/test_model_training.py

```diff
@@ -43,8 +43,6 @@
 def test_get_total_number_of_training_epochs() -> None:
     c = DeepLearningConfig(num_epochs=2, should_validate=False)
     assert c.get_total_number_of_training_epochs() == 2
-    c = DeepLearningConfig(num_epochs=10, start_epoch=5, should_validate=False)
-    assert c.get_total_number_of_training_epochs() == 5
```

Comment:
You could still have that assert in place, now working with the _start_epoch field that you still use.